### PR TITLE
feat: Allow non-batched inputs on indexed_add and indexed_put

### LINKED ISSUE #1212: Allow non-batched inputs on indexed_add and indexed_put
Instead of `{n, rank(t)}`, we can give `{rank(t)}`. If a scalar is given, it is the same as `{1}`.

### PR DESCRIPTION
**Pull request by polvalente**
closes #1212

### CODE REVIEW

File: nx/lib/nx.ex [hunk 1]
```diff
@@ -7387,6 +7409,12 @@ defmodule Nx do
     indexed_op(target, indices, updates, :indexed_put)
   end
 
+  defp indexed_op(target, %Nx.Tensor{shape: {_}} = index, update, op) when is_tensor(update) do
```
Comment (polvalente) [inline nx/lib/nx.ex:7412]: @josevalim in theory we could extend this to accept a list of numbers/tensors as well in the `index`. WDYT?

Would be something like `index = if is_list(index), do: Enum.map(index, &to_tensor/1) |> Nx.concatenate(), else: to_tensor(index)`

**Comment by josevalim** [comment]
Can we also support scalars? Otherwise it looks good to me. I also want to rewrite the docs, to make the simpler path come first, but I can do that after merging. :)

**Comment by polvalente** [comment]
What do you mean by "also support scalars"?

**Comment by josevalim** [comment]
I mean this: `Nx.indexed_put(Nx.tensor([1, 2, 3]), Nx.tensor(1), 10)`.

**Pull request merged**